src/net_processing.cpp

```diff
@@ -1395,7 +1395,8 @@ void PeerManagerImpl::PushNodeVersion(CNode& pnode, const Peer& peer)
 
 void PeerManagerImpl::AddTxAnnouncement(const CNode& node, const GenTxid& gtxid, std::chrono::microseconds current_time)
 {
-    AssertLockHeld(::cs_main); // For m_txrequest
+    AssertLockHeld(::cs_main);
```

Comment:
Not sure if this is "correct". While `m_txrequest` has an internal mutex to guard against UB, the internal mutex does nothing to guard the processing logic. A mutex different from the internal one is still needed here to guard against several threads calling into this function at the same time.

Reply:
Hm I think you're right. Also seems similar to what Vasil [mentioned](https://github.com/bitcoin/bitcoin/pull/26151#issuecomment-1286897751), when we call methods on `m_txrequest` but expect the internal state between those calls not to change, then the internal mutex won't help us. afaict the changes here don't break anything because `cs_main` is still held in these places but I am trying to seperate `m_txrequest` from `cs_main` so `cs_main` should not be needed for any of this after the PR. Will mark as draft until I figure out how to address this. (Seems like the only way to address this is to change the `TxRequestTracker` interface)

Reply:
Yeah, an alternative would be to replace cs_main with g_msgproc_mutex?

Reply:
Or expose the mutex outside of `TxRequestTracker` and lock for longer duration in the net processing code. I.e. have `TxRequestTracker m_txrequest GUARDED_BY(m_its_own_mutex);` as suggested by @ajtowns in https://github.com/bitcoin/bitcoin/pull/26151#pullrequestreview-1116661944